src/lib/arch/aarch64/Architecture.cc

```diff
@@ -26,15 +22,23 @@ Architecture::Architecture(kernel::Linux& kernel, YAML::Node config)
 
   cs_option(capstoneHandle, CS_OPT_DETAIL, CS_OPT_ON);
 
+  // Initialise SVE and SME vector lengths
+  ryml::ConstNodeRef config = config::SimInfo::getConfig();
+  config["Core"]["Vector-Length"] >> VL_;
+  config["Core"]["Streaming-Vector-Length"] >> SVL_;
+
+  // Initialise virtual counter timer increment frequency
+  float clkFreq;
+  config["Core"]["Clock-Frequency"] >> clkFreq;
+  uint32_t timFreq;
+  config["Core"]["Timer-Frequency"] >> timFreq;
+  vctModulo_ = (clkFreq * 1e9) / (timFreq * 1e6);
+
   // Generate zero-indexed system register map
-  systemRegisterMap_[ARM64_SYSREG_DCZID_EL0] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_FPCR] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_FPSR] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_TPIDR_EL0] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_MIDR_EL1] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_CNTVCT_EL0] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_PMCCNTR_EL0] = systemRegisterMap_.size();
-  systemRegisterMap_[ARM64_SYSREG_SVCR] = systemRegisterMap_.size();
+  std::vector<uint64_t> sysRegs = config::SimInfo::getSysRegVec();
+  for (size_t i = 0; i < sysRegs.size(); i++) {
+    systemRegisterMap_[sysRegs[i]] = systemRegisterMap_.size();
+  }
```

Comment:
What is the need for this map given `getSysRegVecIndex()` in SimInfo?

Reply:
`getSysRegVecIndex` is currently unused (perhaps should be removed?) as the getting of the sys reg tag has to be done without a critical failure to the simulation. Incorrectly speculated instructions can possibly generate an invalid sys reg tag

Reply:
Given it is unused then I would say that `getSysRegVecIndex` should be removed